### DISCUSSION TITLE
Improve remix-architect isBinaryType performance

### DISCUSSION THREAD

**Discussion by gyx1000** [remix-run/remix#4760] [Proposals]
I'm starting to explore the beautiful remix project.

I noticed that the isBinaryType function of the remix-architect adapter has a complexity of O(N*includesComplexity)

Would it be possible to convert the binaryTypes list to Set and split the contentType variable by ; (relative to rfc7231) and do something like that to have O(1) complexity?

```typescript
export function isBinaryType(contentType: string | null | undefined) {
  if (!contentType) return false;
  let [test, ] = contentType.split(';');
  return binaryTypes.has(test);
}
```

Thanks again for this wonderful project.

**Comment by machour** [comment]
@gyx1000 Welcome to the community 🙌
Could you open a PR targeting the `dev` branch with your changes? It would make it way easier for the team to review.

**Reply by gyx1000** [reply]
Hi @machour,

I opened #4761

**Reply by machour** [reply]
Thanks! We just need you to sign the CLA by adding your name to the contributors.yml file and we'll be good to go.
